### PR TITLE
Replace `wasm-run` by `xtask-wasm`

### PR DESCRIPTION
**Pull request by yozhgoor**

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -0,0 +1,1757 @@
+# This file is automatically @generated by Cargo.
```
Comment (cecton) [inline Cargo.lock]: Oh, there is a cargo.lock now? xD

File: xtask/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+[package]
+name = "xtask"
+version = "0.1.0"
+edtion = "2021"
+authors = ["{{authors}}"]
```
Comment (cecton) [inline xtask/Cargo.toml]: I think we don't need this at all anymore
```suggestion
```

**Review by cecton — APPROVED** [review]

File: .genignore [hunk 3]
```diff
@@ -1,2 +1 @@
-.cargo-ok
```
Comment (cecton) [inline .genignore:1]: I don't remember what this file does. And why are you removing this?

File: .gitignore [hunk 4]
```diff
@@ -1,3 +1,2 @@
 /target
-/build
-/app/static/blueprint.css
+static/blueprint.css
```
Comment (cecton) [inline .gitignore]:
```suggestion
/static/blueprint.css
```

File: xtask/src/main.rs [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+fn download_css(force: bool) -> Result<()> {
+    let static_path = PathBuf::from("static");
+    let css_path = static_path.join("blueprint.css");
+
+    if force || !css_path.exists() {
+        yewprint_css::download_css(&css_path)?;
+    }
+
+    Ok(())
+}
```
Comment (cecton) [inline xtask/src/main.rs]: I don't think this should be download into the static directory but into the dist directory instead. wdyt?

File: .genignore (hunk 3 above)
Comment (yozhgoor) [reply .genignore:1]: `cargo-generate` ignore these files by default (https://github.com/cargo-generate/cargo-generate/pull/269)

File: .github/workflows/main.yml [hunk 6]
```diff
@@ -16,19 +16,10 @@ jobs:
...
-        id: cache-target
-        uses: actions/cache@v2
-        with:
-          path: |
-            ~/.rustup
-            ~/.cargo/registry
-            ~/.cargo/bin
-            target
-          key: main
+      - uses: Swatinem/rust-cache@v1
```
Comment (cecton) [inline .github/workflows/main.yml:19]: :+1:

File: .github/workflows/main.yml [hunk 7]
```diff
@@ -16,19 +16,10 @@ jobs:
...
 
       - name: cargo test
         uses: actions-rs/cargo@v1
         with:
           command: test
-          args: --workspace
+          args: --workspace --all-features
```
Comment (cecton) [inline .github/workflows/main.yml:25]: :+1:

File: Cargo.toml [hunk 8]
```diff
@@ -1,26 +1,19 @@
...
+[lib]
+crate-type = ["cdylib"]
 
-# Helper to download blueprint's CSS
-yewprint-css = "0.1.0"
+[dependencies]
+wasm-bindgen = "0.2"
+web-sys = { version = "0.3", features = ["Window", "MediaQueryList"] }
+yew = "0.19"
+yewprint = { git = "https://github.com/yewprint/yewprint.git", branch = "main" }
```
Comment (cecton) [inline Cargo.toml]: Shouldn't we use the stable release??

File: Cargo.toml [hunk 9]
```diff
@@ -1,26 +1,19 @@
...
 
 [workspace]
-# Our actual web app
-members = ["app"]
-
-[profile.release]
-# Tell `rustc` to optimize for small code size.
-opt-level = "s"
```
Comment (cecton) [inline Cargo.toml]: Pretty sure we need to keep this here for optimizing the wasm

File: LICENSE.Apache-2.0 [hunk 10]
```diff
@@ -1,176 +1,201 @@
...
+      To apply the Apache License to your work, attach the following
+      boilerplate notice, with the fields enclosed by brackets "[]"
+      replaced with your own identifying information. (Don't include
+      the brackets!)  The text should be enclosed in the appropriate
+      comment syntax for the file format. We also recommend that a
+      file or class name and description of purpose be included on the
+      same "printed page" as the copyright notice for easier
+      identification within third-party archives.
+
+   Copyright 2022 {{authors}}
```
Comment (cecton) [inline LICENSE.Apache-2.0]: There isnt a placeholder in cargo generate for the year?

File: LICENSE.MIT [hunk 11]
```diff
@@ -1,25 +1,21 @@
...
-Permission is hereby granted, free of charge, to any
-person obtaining a copy of this software and associated
-documentation files (the "Software"), to deal in the
-Software without restriction, including without
-limitation the rights to use, copy, modify, merge,
-publish, distribute, sublicense, and/or sell copies of
-the Software, and to permit persons to whom the Software
-is furnished to do so, subject to the following
-conditions:
+Copyright (c) 2022 {{authors}}
```
Comment (cecton) [inline LICENSE.MIT]: (Year again)

File: src/app.rs [hunk 12]
```diff
@@ -1,9 +1,15 @@
...
+pub fn root() -> Html {
+    html! {
+        <App></App>
+    }
+}
+
 pub struct App {
     dark_theme: bool,
-    link: ComponentLink<Self>,
 }
```
Comment (cecton) [inline src/app.rs]: 1. You could/should probably rewrite App to the functional component
2. I'm not sure why you introduced this Root component??

File: xtask/src/main.rs [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+    Watch(xtask_wasm::Watch),
+    Start(xtask_wasm::DevServer),
+}
+
+fn main() -> Result<()> {
+    let cli: Cli = clap::Parser::parse();
+
+    match cli {
+        Cli::Dist(args) => {
+            let dist = args.static_dir_path("static").run("{{project-name}}")?;
```
Comment (cecton) [inline xtask/src/main.rs]: This was meant to be used this way:

```suggestion
            let DistResult { dist_dir, .. } = args.static_dir_path("static").run("{{project-name}}")?;
```

File: xtask/src/main.rs [hunk 14]
```diff
@@ -0,0 +1,44 @@
...
+        Cli::Start(args) => {
+            args.arg("dist")
+                .start(xtask_wasm::default_dist_dir(false))?;
+        }
+    }
+
+    Ok(())
+}
+
+fn download_css(path: PathBuf) -> Result<()> {
```
Comment (cecton) [inline xtask/src/main.rs]:
```suggestion
fn download_css(path: &Path) -> Result<()> {
```

**Pull request merged**